api/submission/fetch_course_report.go

```diff
@@ -0,0 +1,30 @@
+package submission
+
+import (
+	"github.com/edulinq/autograder/api/core"
```

Comment:
Inconsistent use of tabs vs spaces in this file.
(This makes it hard to tell if your indentation is messed up in some parts.)

This may be true of other files, but I will only note it once.